### PR TITLE
Restore coverage in Travis

### PR DESCRIPTION
**Pull request by vjeux**
We disabled coverage in Travis because the implementation was crashing ( https://github.com/facebook/react/issues/6290 ). Since we upgraded to Jest 15, the entire coverage implementation is brand new so we should give it another try.

### CODE REVIEW

File: grunt/tasks/jest.js [hunk 1]
```diff
@@ -1,84 +1,17 @@
 'use strict';
 
-var async = require('async');
 var fs = require('fs');
-var glob = require('glob');
 var grunt = require('grunt');
 var path = require('path');
 
-var rootPath = path.resolve('.');
```
Comment (vjeux) [inline grunt/tasks/jest.js]: All this code path was really weird and not needed anymore. We can just run `jest --coverage` to do the same thing.

File: grunt/tasks/jest.js [hunk 2]
```diff
@@ -1,84 +1,17 @@
...
-}
-
-function run(done, configPath) {
+function run(done, coverage) {
   grunt.log.writeln('running jest');
 
   var args = [
     path.join('node_modules', 'jest', 'bin', 'jest'),
-    '--runInBand',
-    '--no-watchman',
```
Comment (vjeux) [inline grunt/tasks/jest.js]: @cpojer told me to try and remove those as jest should be more robust. I'll put them back if they cause issues.

File: package.json [hunk 3]
```diff
@@ -58,7 +58,7 @@
     "gulp-load-plugins": "^1.2.4",
     "gulp-util": "^3.0.7",
     "gzip-js": "~0.3.2",
-    "jest": "^15.0.1",
+    "jest": "^15.0.2",
```
Comment (vjeux) [inline package.json]: This fixed a bug with coverage and node 4

File: .travis.yml [hunk 4]
```diff
@@ -71,14 +71,8 @@ script:
...
-      # Disabling coverage because it's broken:
-      # https://travis-ci.org/facebook/react/jobs/128163922
-      if false; then
-        ./node_modules/.bin/grunt jest:coverage
-        cat ./coverage/lcov.info | ./node_modules/.bin/coveralls
-      else
-        ./node_modules/.bin/grunt jest:normal
-      fi
+      ./node_modules/.bin/grunt jest:coverage
+      cat ./coverage/lcov.info | ./node_modules/.bin/coveralls
```
Comment (cpojer) [inline .travis.yml]: Is coveralls better than codecov? For Jest we now use Codecov and it comments on every PR with the coverage difference. It is nice.

**Comment by cpojer** [comment]
oh man, this is so much simpler.

Does this run coverage on every PR or only on master? I think we should start collecting coverage on PRs too, possibly with codecov commenting on diffs.

If this passes on travis just fine, shall we merge it and see if it works well on PRs over the course of a day and potentially revert if there is trouble? I don't expect there to be any, but you know, all libraries have different, crazy tests. Overall, the Jest tests are very similar to React though.

File: .travis.yml (hunk 4 above)
Comment (Aweary) [inline .travis.yml]: coveralls should do that too https://coveralls.io/features

**Comment by cpojer** [comment]
omg it works.

**Comment by vjeux** [comment]
Do you guys know if it's normal that it says "Changes Unknown" but manages to get the correct 67%?

Maybe it's because it's the first time it runs?

File: grunt/tasks/jest.js (hunk 1 above)
Comment (zpao) [inline grunt/tasks/jest.js]: I don't think that's true. We want to exclude some files for sure, eg https://github.com/facebook/react/tree/355c49065386cbe33c026ba573c88c4e459ea328/src/shared/vendor/third_party (that file is showing up in the coverage report for this build). I'm also seeing coverage results for scripts/jest/test-framework-setup.js in the report.

**Comment by gaearon** [comment]
Could be

**Comment by zpao** [comment]
Yea, probably fine. I'm guessing it's looking at the branch point and trying to find coverage for that commit but since we haven't had coverage on for a while it cant find anything.

**Comment by vjeux** [comment]
Okay, here's my suggestion: we merge it in and see how it works in the next few prs that come in.

File: grunt/tasks/jest.js (hunk 1 above)
Comment (vjeux) [inline grunt/tasks/jest.js]: Jest has this annoying mode where you can either use the default where it runs coverage on all the files that are required by tests (the current mode) or you can manually specify all the files/folders you want to whitelist and blacklist.

My thinking was that outside of this thirty party component file, the rest is actually good. When I tried doing it on everything then it reported a bunch of useless stuff.

By the way, we don't need all that logic anymore, jest has a setting `collectCoverageFrom` in package.json that lets us configure it: https://github.com/facebook/jest/blob/master/package.json#L47

If you want to go through all the files and make it list only the things we care about, let me know.

**Comment by zpao** [comment]
Please make sure you exclude files appropriately (I think my comment above may be getting lost since it's inline).

File: package.json [hunk 5]
```diff
@@ -111,6 +111,12 @@
       "<rootDir>/src",
       "node_modules/fbjs"
     ],
+    "collectCoverageFrom": [
```
Comment (vjeux) [inline package.json]: @zpao I put back the old coverage paths

**Comment by vjeux** [comment]
Yay, 87%

**Comment by vjeux** [comment]
Thanks!

**Pull request merged**